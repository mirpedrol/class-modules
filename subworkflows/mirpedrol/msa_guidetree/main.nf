if ( params.guidetree == "clustalo/guidetree" ) {
    include { CLUSTALO_GUIDETREE as GUIDETREE } from '../../../modules/mirpedrol/clustalo/guidetree/main'
} else if ( params.guidetree == "famsa/guidetree" ) {
    include { FAMSA_GUIDETREE    as GUIDETREE } from '../../../modules/mirpedrol/famsa/guidetree/main'
} else if ( params.guidetree == "magus/guidetree" ) {
    include { MAGUS_GUIDETREE    as GUIDETREE } from '../../../modules/mirpedrol/magus/guidetree/main'
}

workflow MSA_GUIDETREE {

    take:
    ch_fasta // channel: [ meta, fasta ]

    main:

    ch_versions = Channel.empty()

    GUIDETREE ( ch_fasta )
    ch_versions = ch_versions.mix(GUIDETREE.out.versions)

    emit:
    guidetree = GUIDETREE.out.tree // channel: [ meta, *.dnd ]
    versions  = ch_versions        // channel: [ versions.yml ]
}

