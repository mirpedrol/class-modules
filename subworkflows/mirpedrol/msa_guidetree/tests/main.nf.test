nextflow_workflow {

    name "Test Subworkflow MSA_GUIDETREE"
    script "../main.nf"
    workflow "MSA_GUIDETREE"

    tag "subworkflows"
    tag "subworkflows_mirpedrol"
    tag "subworkflows/msa_guidetree"
    tag "clustalo/guidetree"
    tag "famsa/guidetree"
    tag "magus/guidetree"

    test("sarscov2 - clustalo/guidetree") {

        when {
            params.guidetree = "clustalo/guidetree"
            workflow {
                """
                input[0] = [ [ id:'test' ], // meta map
                             file(params.test_data['sarscov2']['illumina']['contigs_fasta'], checkIfExists: true)
                           ]
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }

    test("sarscov2 - famsa/guidetree") {

        when {
            params.guidetree = "famsa/guidetree"
            workflow {
                """
                input[0] = [ [ id:'test' ], // meta map
                             file(params.test_data['sarscov2']['illumina']['contigs_fasta'], checkIfExists: true)
                           ]
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }

    test("sarscov2 - magus/guidetree") {

        when {
            params.guidetree = "magus/guidetree"
            workflow {
                """
                input[0] = [ [ id:'test' ], // meta map
                             file("https://raw.githubusercontent.com/nf-core/test-datasets/multiplesequencealign/testdata/setoxin-ref.fa", checkIfExists: true)
                           ]
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert path(workflow.out.guidetree[0][1]).getText().contains("1apf") },
                { assert path(workflow.out.guidetree[0][1]).getText().contains("1ahl") },
                { assert path(workflow.out.guidetree[0][1]).getText().contains("1atx") },
                { assert path(workflow.out.guidetree[0][1]).getText().contains("1sh1") },
                { assert path(workflow.out.guidetree[0][1]).getText().contains("1bds") },
                { assert snapshot(workflow.out.versions).match("versions1") }
            )
        }
    }
}
