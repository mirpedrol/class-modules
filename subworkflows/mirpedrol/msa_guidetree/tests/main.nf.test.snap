{
    "sarscov2 - famsa/guidetree": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.dnd:md5,f3ef8b16a7a16cb4548942ebf2e7bad6"
                    ]
                ],
                "1": [
                    "versions.yml:md5,723b3358beebc97847b4681f562bcea0"
                ],
                "guidetree": [
                    [
                        {
                            "id": "test"
                        },
                        "test.dnd:md5,f3ef8b16a7a16cb4548942ebf2e7bad6"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,723b3358beebc97847b4681f562bcea0"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-16T13:12:32.7754812"
    },
    "sarscov2 - clustalo/guidetree": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.dnd:md5,5428bad500a0a0bd985744bec1a12a70"
                    ]
                ],
                "1": [
                    "versions.yml:md5,b1ee3efbf09bc7cf7b4970916a00fddc"
                ],
                "guidetree": [
                    [
                        {
                            "id": "test"
                        },
                        "test.dnd:md5,5428bad500a0a0bd985744bec1a12a70"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b1ee3efbf09bc7cf7b4970916a00fddc"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-16T13:12:06.216087789"
    },
    "versions1": {
        "content": [
            [
                "versions.yml:md5,75333144e16039f25cae8e933f30d003"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-16T13:13:24.823000398"
    }
}